python/src/iceberg/types.py

```diff
@@ -305,10 +364,10 @@ class DoubleType(Type, Singleton):
     """
 
     def __init__(self):
-        super().__init__("double", "DoubleType()", is_primitive=True)
+        super().__init__("double", "DoubleType()")
 
 
-class DateType(Type, Singleton):
+class DateType(IcebergType, Singleton):
```

Comment:
These all need to be `PrimitiveType` to make the `is_primitive` check work, right?